codex-rs/core/src/windows.rs

```diff
@@ -0,0 +1,250 @@
+#[cfg(target_os = "windows")]
+use crate::config_types::SandboxWorkspaceWrite;
+use crate::config_types::WindowsConfig;
+use crate::exec::SandboxType;
+use crate::protocol::SandboxPolicy;
+use crate::spawn::StdioPolicy;
+#[cfg(target_os = "windows")]
+use crate::spawn::spawn_child_async;
+use std::collections::HashMap;
+use std::io;
+use std::path::Path;
+use std::path::PathBuf;
+use tokio::process::Child;
+
+#[cfg(target_os = "windows")]
+use std::process::Command as StdCommand;
+#[cfg(target_os = "windows")]
+use std::sync::OnceLock;
+#[cfg(target_os = "windows")]
+use std::sync::RwLock;
+#[cfg(target_os = "windows")]
+use std::sync::atomic::AtomicBool;
+#[cfg(target_os = "windows")]
+use std::sync::atomic::Ordering;
+
+#[cfg(target_os = "windows")]
+static WINDOWS_SETTINGS: RwLock<WindowsConfig> = RwLock::new(WindowsConfig::default());
+#[cfg(target_os = "windows")]
+static NOTICE_PRINTED: AtomicBool = AtomicBool::new(false);
+#[cfg(target_os = "windows")]
+static WSL_AVAILABLE: OnceLock<bool> = OnceLock::new();
+
+pub fn update_settings(settings: WindowsConfig) {
+    #[cfg(target_os = "windows")]
+    {
+        if let Ok(mut guard) = WINDOWS_SETTINGS.write() {
+            *guard = settings;
+        }
+    }
+    #[cfg(not(target_os = "windows"))]
+    {
+        let _ = settings;
+    }
+}
+
+pub fn preferred_sandbox() -> Option<SandboxType> {
+    if prefer_wsl() {
+        Some(SandboxType::WindowsWsl)
+    } else {
+        None
+    }
+}
+
+pub fn prefer_wsl() -> bool {
```

Comment:
I don't think this should be a global function: this should be derived from an instance of `Config`.